### DISCUSSION TITLE
How to access ws->getUserData in listen()

### DISCUSSION THREAD

**Discussion by mmoelle1** [uNetworking/uWebSockets#1609] [Q&A]
Dear community,

is it possible to access `ws->getUserData` in `.listen(...)`? I consider `listen` to be the "constructor" where I can do initialisation tasks that need to be done just once (not at every `open()` request). I need to pass command-line arguments, that's why I thought `listen()` would be the right place.

**Comment by uNetworkingAB** [comment]
You should read the user manual and docs.

> I consider listen to be the "constructor"

Of the **app**, yes. Of **websockets**, no.

**Reply by uNetworkingAB** [reply]
Hmm, maybe you talk about the prototype of the JS object that appears in ws->getUserData? There is no prototype, it's just an empty object.

**Reply by mmoelle1** [reply]
> You should read the user manual and docs.
> 
> > I consider listen to be the "constructor"
> 
> Of the **app**, yes. Of **websockets**, no.

I should have been more specific here, I am using the C++ API. I actually had looked at the user manual and docs before but could not find a sufficient answer in there. I know how to access `ws->getUserData` in, say, `.message = [](auto *ws, ...)` or `.open = [](auto *ws)` because `ws` is passed there as a pointer. 

However, the signature of listen is like this
```
.listen(port_value, [](auto* listen_socket)
```
whereby `listen_socket` is (to the best of my knowledge) of type `us_listen_socket_t`. Any help on how to arrive at `ws` from `listen_socket` is appreciated.

**Reply by uNetworkingAB** [reply]
The listen socket doesn't hold or give you WebSockets, you get them in **open**
